tests/openedx_tagging/core/tagging/test_views.py

```diff
@@ -0,0 +1,365 @@
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.put(
+            url, {"name": "new name", "system_defined": update_value}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            assert response.data["system_defined"] == False
+
+    def test_update_taxonomy_404(self):
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": 123123})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.put(url, {"name": "new name"}, format="json")
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @ddt.data(
+        (None, status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_403_FORBIDDEN),
+        ("staff", status.HTTP_200_OK),
+    )
+    @ddt.unpack
+    def test_patch_taxonomy(self, user_attr, expected_status):
+        taxonomy = Taxonomy.objects.create(
+            name="test patch taxonomy", enabled=False, required=True
+        )
+        taxonomy.save()
+
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        if user_attr:
+            user = getattr(self, user_attr)
+            self.client.force_authenticate(user=user)
+
+        response = self.client.patch(
+            url, {"name": "new name", "required": False}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        # If we were able to update the taxonomy, check if the name changed
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            check_taxonomy(
+                response.data,
+                response.data["id"],
+                **{
+                    "name": "new name",
+                    "enabled": False,
+                    "required": False,
+                },
+            )
+
+    @ddt.data(
+        (False, False, status.HTTP_200_OK),
+        (False, True, status.HTTP_200_OK),
+        (True, False, status.HTTP_403_FORBIDDEN),
+        (True, True, status.HTTP_403_FORBIDDEN),
+    )
+    @ddt.unpack
+    def test_patch_taxonomy_system_defined(
+        self, create_value, update_value, expected_status
+    ):
+        taxonomy = Taxonomy.objects.create(
+            name="test system taxonomy", system_defined=create_value
+        )
+        taxonomy.save()
+        url = reverse("oel_tagging:taxonomy-detail", kwargs={"pk": taxonomy.pk})
+
+        self.client.force_authenticate(user=self.staff)
+        response = self.client.patch(
+            url, {"system_defined": update_value}, format="json"
+        )
+        assert response.status_code == expected_status
+
+        if status.is_success(expected_status):
+            response = self.client.get(url)
+            assert response.data["system_defined"] == False
```

Comment:
Could you make this test clearer with:

```suggestion
        # Verify that system_defined has not changed
        response = self.client.get(url)
        assert response.data["system_defined"] == create_value
```